sdks/python/apache_beam/transforms/core.py

```diff
@@ -165,13 +190,72 @@ def default_label(self):
     return self.__class__.__name__
 
   def process(self, element, *args, **kwargs):
-    """Called for each element of a pipeline. The default arguments are needed
-    for the DoFnRunner to be able to pass the parameters correctly.
+    """method to use for processing elements.
+
+    This is invoked by ``DoFnRunner`` for each element of a input
+    ``PCollection``.
+
+    If specified, following default arguments are used by the ``DoFnRunner`` to
+    be able to pass the parameters correctly.
+
+    ``DoFn.ElementParam``: element to be processed.
+    ``DoFn.SideInputParam``: a side input that may be used when processing.
+    ``DoFn.TimestampParam``: timestamp of the input element.
+    ``DoFn.WindowParam``: ``Window`` the input element belongs to.
+    ``DoFn.RestrictionTrackerParam``: a ``RestrictionTrakcer`` to be used when
+    processing the input element. If a parameter is defined with this default
+    value, current ``DoFn`` is considered to be a Splittable ``DoFn`` and is
+    subjected to the special requirements given in the corresponding section
+    below.
+    ``DoFn.WatermarkReporterParam``: a function that can be used to give the
+    runner a (best-effort) lower bound about the timestamps of future output
+    associated with the current element. If the ``DoFn`` has multiple outputs,
+    the watermark applies to all of them. Function must be invoked with a
+    single parameter of type ``Timestamp`` or as an integer that gives the
+    watermark in number of seconds.
+
+    ** Splittable DoFns **
+    A ``DoFn`` is considered to be a 'Splittable DoFn' of it's ``process()``
+    method has a parameter with default value ``DoFn.RestrictionTrackerParam``.
+    This is an advanced feature and an overwhelming majority of users will never
+    need to write a Splittable DoFn.
+
+    Not all runners support Splittable DoFn. See the capability matrix
+    (a href="https://beam.apache.org/documentation/runners/capability-matrix/)
```

Comment:
The matrix is currently focused on Java, so this is a bit misleading.

Reply:
Removed this paragraph.